iFixit/lighthouse-docker
# Create python script to run Lighthouse and send metrics to DataDog

## Description

This is a similar approach to the DataDog Lighthouse Integration seen here https://github.com/DataDog/integrations-extras/blob/master/lighthouse/datadog_checks/lighthouse/lighthouse.py. 

Most of the logic is in the `main.py` script which will iterate through a constant list of URLs, run Lighthouse on each URL, parse the results for core metrics, and send these metrics over to DataDog using the wrapper class.

In future pulls we attempt to run this as a cron task to constantly audit the list of URLs and send metrics to DataDog.

### CR

With this approach, we will be able to retrieve the network request size which will help to address the original issue described in https://github.com/iFixit/ifixit/discussions/47449.

At the moment, I decided to only send the cumulative layout shift, largest contentful paint, total blocking time, and page size. The first three metrics are weighted the most by Google. 


### QA

1. Run `pip install -r /path/to/requirements.txt` to install the required dependencies
2. Have `lighthouse` installed locally
3. Run `python3 lib/main.py` and confirm that lighthouse was run for each URL and the metrics show up in DataDog

**cc:** @andyg0808 
Closes: https://github.com/iFixit/ifixit/issues/48149
